core/src/main/java/io/grpc/internal/DnsNameResolverProvider.java

```diff
@@ -38,37 +31,11 @@
  *   <li>{@code "dns:///foo.googleapis.com"} (without port)</li>
  * </ul>
  */
-public final class DnsNameResolverProvider extends NameResolverProvider {
-
-  private static final String SCHEME = "dns";
-
-  @Override
-  public DnsNameResolver newNameResolver(URI targetUri, NameResolver.Args args) {
-    if (SCHEME.equals(targetUri.getScheme())) {
-      String targetPath = Preconditions.checkNotNull(targetUri.getPath(), "targetPath");
-      Preconditions.checkArgument(targetPath.startsWith("/"),
-          "the path component (%s) of the target (%s) must start with '/'", targetPath, targetUri);
-      String name = targetPath.substring(1);
-      return new DnsNameResolver(
-          targetUri.getAuthority(),
-          name,
-          args,
-          GrpcUtil.SHARED_CHANNEL_EXECUTOR,
-          Stopwatch.createUnstarted(),
-          InternalServiceProviders.isAndroid(getClass().getClassLoader()));
-    } else {
-      return null;
-    }
-  }
-
-  @Override
-  public String getDefaultScheme() {
-    return SCHEME;
-  }
+public final class DnsNameResolverProvider extends BaseDnsNameResolverProvider {
 
   @Override
-  protected boolean isAvailable() {
-    return true;
+  protected boolean isSrvEnabled() {
+    return Boolean.parseBoolean(System.getProperty(ENABLE_GRPCLB_PROPERTY_NAME, "false"));
```

Comment:
The original code parses this boolean in a static context and assign it to a static field. We should keep doing this to prevent the original behavior.